crates/ruma-federation-api/src/membership.rs

```diff
@@ -1,7 +1,175 @@
 //! Room membership endpoints.
 
+use ruma_common::serde::{from_raw_json_value, JsonCastable, JsonObject, Raw};
+use ruma_events::{AnyStrippedStateEvent, PossiblyRedactedStateEventContent, StrippedStateEvent};
+use serde::{Deserialize, Serialize};
+use serde_json::value::RawValue as RawJsonValue;
+
 pub mod create_invite;
 pub mod create_join_event;
+pub mod create_knock_event;
 pub mod create_leave_event;
 pub mod prepare_join_event;
+pub mod prepare_knock_event;
 pub mod prepare_leave_event;
+
+/// Possible formats of events that may appear in place of stripped state events.
```

Comment:
What do you think of:

```suggestion
/// Possible event formats that may appear in stripped state.
```
?